func/mpi/mpi_onesided.cpp

```diff
@@ -44,9 +44,9 @@ FAASM_MAIN_FUNC()
     int* sharedData;
     MPI_Alloc_mem(memSize, MPI_INFO_NULL, &sharedData);
 
-    int putData[NUM_ELEMENT];
-    int expectedGetData[NUM_ELEMENT];
-    int expectedPutData[NUM_ELEMENT];
+    int* putData = new int[NUM_ELEMENT];
```

Comment:
These should all be `auto` (i.e. avoid explicitly typing pointers that come from `new` like this)